### PR TITLE
TAS: fix scenario when parallelism < completions

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Part of #3450 

#### Does this PR introduce a user-facing change?

```release-note
Fix running Job when parallelism < completions, before the fix the replacement pods for the successfully
completed Pods were not ungated.
```

### CODE REVIEW

File: test/e2e/tas/tas_test.go [hunk 1]
```diff
@@ -233,6 +233,32 @@ var _ = ginkgo.Describe("TopologyAwareScheduling", func() {
 				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
 			})
 		})
+
+		ginkgo.It("Should allow to run a Job with parallelism < completions", func() {
+			sampleJob := testingjob.MakeJob("test-job", ns.Name).
+				Queue(localQueue.Name).
+				Parallelism(3).
```
Comment (mbobrovskyi) [inline test/e2e/tas/tas_test.go:240]:
```suggestion
				Parallelism(2).
```
?

Comment (mimowo) [reply test/e2e/tas/tas_test.go]: thx, sure

**Comment by mbobrovskyi** [comment]
/lgtm
Thanks!

**Pull request merged**